File: mySeriesList/src/main/java/com/msl/myserieslist/entities/Watchlist.java
package com.msl.myserieslist.entities;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import javax.persistence.*;

@Getter
@Setter
@EqualsAndHashCode
@ToString
@Entity
@Table(name = "watchlist", schema = "myserieslist")
public class Watchlist {
    public enum Status {WATCHING, COMPLETED, DROPPED}

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private int id;

    @ManyToOne
    @JoinColumn(name = "idUser", nullable = false)
    private User user;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "idSeries", nullable = false)
    private Series series;

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "status")
    private Status status;

    @Basic
    @Column(name = "progress", nullable = false)
    private int progress;

    @Basic
    @Column(name = "score")
    private int score;

    @Basic
    @Column(name = "comment")
    private String comment;
}

File: MyWatchlist/MyWatchlistAndroid/app/src/main/java/it/niko/mywatchlistandroid/model/Watchlist.kt
package it.niko.mywatchlistandroid.model

import com.google.gson.annotations.SerializedName

data class Watchlist  (
    @SerializedName("series")
    var series: Series,

    @SerializedName("status")
    var status: Status,

    @SerializedName("progress")
    var progress: Int,

    @SerializedName("score")
    var score: Int,

    @SerializedName("comment")
    var comment: String
)
